rust-escrow/stable-escrow/src/escrow.rs

```diff
@@ -0,0 +1,275 @@
+            "on_withdraw_callback".to_string(),
+            json!({
+                "receiver_id": receiver_id.to_string(),
+                "amount": amount.to_string(),
+            })
+            .to_string()
+            .into_bytes(),
+            0,
+            GAS_FT_TRANSFER_CB,
+        );
+
+        promise.then(callback)
+    }
+
+    #[payable]
+    pub fn claim_fees(&mut self) -> Promise {
+        assert_one_yocto();
+
+        if self.is_deposit_allowed() {
+            env::panic_str("ERR_CLAIM_FEES_NOT_ALLOWED");
+        }
+
+        if self.get_fees().claimed {
+            env::panic_str("ERR_FEES_ALREADY_CLAIMED");
+        }
+
+        let mut fees_to_collect = self.get_fees().amount;
+
+        if !self.is_deposit_allowed() && !self.is_withdrawal_allowed() {
+            // If the escrow is successful
+            // Half of the fees are collected and half invested in the asset
+            fees_to_collect = (self.get_fees().amount as f32 * 0.5) as Balance;
+        }
+
+        let receiver_id = self.get_fees().account_id.clone();
+
+        let promise = Promise::new(self.get_metadata().nep_141.clone()).function_call(
+            "ft_transfer".to_string(),
+            json!({
+                "amount": fees_to_collect.to_string(),
+                "receiver_id": receiver_id.to_string(),
+            })
+            .to_string()
+            .into_bytes(),
+            1, // 1 yoctoNEAR
+            GAS_FT_TRANSFER,
+        );
+
+        let fees_to_invest = self
+            .get_fees()
+            .amount
+            .checked_sub(fees_to_collect)
+            .unwrap_or_else(|| env::panic_str("ERR_FEES_TO_INVEST_OVERFLOW"));
+
+        let callback = Promise::new(env::current_account_id()).function_call(
+            "on_claim_fees_callback".to_string(),
+            json!({
+                "fees_to_invest": fees_to_invest.to_string(),
+            })
+            .to_string()
+            .into_bytes(),
+            0,
+            GAS_FT_TRANSFER_CB,
+        );
+
+        promise.then(callback)
+    }
+
+    /**
+     * Only if total funds are reached, allow to call this function
+     * Transfer total NEP141 funds to a new DAO
+     * Make the depositors members of the DAO
+     */
+    #[payable]
+    pub fn delegate_funds(&mut self, amount: Option<U128>) -> Promise {
+        assert_one_yocto();
+
+        if self.is_deposit_allowed() || self.is_withdrawal_allowed() {
+            env::panic_str("ERR_DELEGATE_NOT_ALLOWED");
+        }
+
+        let receiver_id = self.get_metadata().maintainer_account_id.clone();
+
+        // Half of the fees are collected and half invested in the asset
+        let fee_collected = (self.get_fees().amount as f32 * 0.5) as Balance;
+
+        // If amount is None then use the funding_amount_limit
+        let amount = amount.unwrap_or(U128(self
+            .get_metadata()
+            .funding_amount_limit));
```

Comment:
If at any time we try to transfer a little more of the funds than we have, then the `ft_transfer` will fail and we will not be able to recover the funds anymore, so a solution is to allow us to specify the `amount` to be transferred. 

I thought in this scenario in case there were to be an error in the calculation of fees that makes the amount to try to withdraw be greater than the amount collected.